vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -186,8 +198,79 @@ def _rpc_call(self, function: str, *args, **kwargs) -> Any:
         self._outgoing.put(request)
         return future.result()
 
-    def process_frame(self, *args, **kwargs) -> DETECTION_NODE_TYPE:
-        return self._rpc_call("process_frame", *args, **kwargs)
+    def process_frame(self,
+                      frame: np.ndarray,
+                      detection_node: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) \
+            -> DETECTION_NODE_TYPE:
+        """Sends a frame to be processed by a worker process.
+        The detection_node input is carefully kept track of, to see if the
+        worker process made any modifications to it.
+        """
+        # Assign individual detection nodes with a _object_id in case they
+        # are modified in the other process, so that they can then be updated
+        input_nodes_by_id: Dict[int, DetectionNode] = {}
+        for node in self._flatten_node(detection_node):
+            node._object_id = id(node)
+            input_nodes_by_id[node._object_id] = node
+
+        # Run the process_frame method
+        # in_nodes is the nodes that were fed as inputs
+        # out_nodes is the nodes the were output by the process_frame method
+        in_nodes, out_nodes = self._rpc_call(
+            "process_frame",
+            frame=frame,
+            detection_node=detection_node,
+            options=options,
+            state=state)
+
+        # Create a flat list of nodes that came from the worker process
+        in_nodes_list = self._flatten_node(in_nodes)
+        out_nodes_list = self._flatten_node(out_nodes)
+
+        # Update any nodes from this process with new information from
+        # the worker process.
+        retval = []
+        for detection_node in in_nodes_list + out_nodes_list:
+            if (not hasattr(detection_node, "_object_id")
+                    or detection_node._object_id not in input_nodes_by_id):
+                # If this is a new detection node, not one that already existed
+                # then there is nothing to 'update'
+                if detection_node in out_nodes_list:
+                    retval.append(detection_node)
+                continue
+
+            input_node = input_nodes_by_id[detection_node._object_id]
+            input_node.attributes.update(detection_node.attributes)
+            input_node.extra_data.update(detection_node.extra_data)
+            if input_node.track_id is None:
+                input_node.track_id = detection_node.track_id
+            if input_node.encoding is None:
+                input_node.encoding = detection_node.encoding
```

Comment:
The above comment, but this time we're talking about encoding.

Reply:
Same. It's the same as a capsule editing the node.encoding in the process_frame function.